### PR TITLE
Requests backend

### PR DESCRIPTION
**Pull request by paulmmaingi**
From previous closed unmerged pr:
>Refactored and modularised authentication code. Now handled by functions in authentication.py .
>**Added email as well as username signup options, yet to implement the email functionality but works with test data and test files.
>Added more security checks for the forms client and server side e.g updated the regex checks and added some custom validators.
>Fixed the .db files on repository, removed and added .db to gitignore.
>**Added a posts db, yet to implement features for it.
>Added an MIT LICENSE.
>Added a requirements.txt file for flask packages and dependencies in the virtual environment.
>**Need to return responses as jsonify objs not use flash.

Checked `login.py` and the functionality is already implemented in the new authentication model. The only thing is I'm returning flash messages at the moment, in new branch might have to change to json so that responses can be captured and rendered in frontend as desired.

Also included my html files that work for the current authentication and a barebones test of the Posts functionality in the `tests` directory.

Some new migrations for the new fields added to models, detailed in tag v1.1 commit.

### CODE REVIEW

**Comment by paulmmaingi** [comment]
Gone through the merge conflicts for main with branch.

File: app/models.py [hunk 1]
```diff
@@ -1,35 +1,56 @@
...
+    image_file = db.Column(db.String(20), nullable=False, default='default.jpg')
+    password = db.Column(db.String(256), nullable=False)
+    # for short bio
+    about = db.Column(db.String(256), default='Hello, I am new here!')
+    # To know when the account was created for requestee and account page
+    account_created = db.Column(db.DateTime, nullable=False, default=datetime.now(timezone.utc))
     ratings = db.Column(db.Float(5.0), default=0.0)
     ratings_count = db.Column(db.Integer(), default=0)
-    # author = db.relationship('Post', backref='posts', lazy='dyanmic')
-
```
Comment (paulmmaingi) [inline app/models.py:13]: UserMixin class already implements these so if i inherit from it no need to explicitly define them.

File: app/routes.py [hunk 2]
```diff
@@ -1,160 +1,30 @@
...
+from .authentication import handle_register, handle_login, handle_logout
 
 
-from flask import request, jsonify, redirect, render_template
-
-from flask_login import LoginManager, current_user, login_user, logout_user
-from werkzeug.security import check_password_hash
-from .forms import LoginForm
-
-
```
Comment (paulmmaingi) [inline app/routes.py:18]: initialisation for this already happening in app module init file

File: app/routes.py [hunk 3]
```diff
@@ -1,160 +1,30 @@
...
-# Login manager
-login_manager = LoginManager()
-login_manager.init_app(flaskApp)
-
-
-@login_manager.user_loader
-def load_user(user_id):
```
Comment (paulmmaingi) [inline app/routes.py:25]: This is in models as its for loading user, thus makes sense in User model file

**Review by paulmmaingi — COMMENTED** [review]
Ok reviewed all the files in this merge again. 
To easily integrate into your local branch the database migrations:
run `flask db downgrade base`
then run `flask db upgrade`
assuming youve already created the database with `flask db init`

**Comment by seanlixu** [comment]
looks good! I'll work on updating the integrating the front end with this shortly, i'll also adopt some of the additional features you've included here. great work!

**Pull request merged**